packages/plugin-core/src/services/EngineAPIService.ts

```diff
@@ -1,12 +1,65 @@
+import {
+  DendronAPI,
+  EngineWriteOptsV2,
+  NoteProps,
+  WriteNoteResp,
+} from "@dendronhq/common-all";
 import { DendronEngineClient, HistoryService } from "@dendronhq/engine-server";
 import path from "path";
 import { DendronWorkspace } from "../workspace";
 
 export class EngineAPIService extends DendronEngineClient {
-  static create({ port }: { port: number | string }) {
```

Comment:
@kevinslin - there's a minor issue here with this static create Factory method.  Because EngineAPIService extends DendronEngineClient, I can't create a similar override `Create` that returns an `EngineAPIService` as the function signature will no longer match.  That's why I've changed the factory method to `createEngine` instead (a bit hacky).  

The static methods don't gel too well with inheritance; any thoughts on reworking the `DendronEngineClient` to use parameterized constructor overloads instead of a static Factory constructor?

Reply:
we've been using `create`to try to give a common interface for different clients. we could parameterize the `create` method as well, having it take an optional generic 

```ts
type CreateOpts<T = any> {
  port: number
  extra?: T
}
```

the idea is that `creating` the client should be fairly specific depending on the client which is why we didn't use inheritance

Reply:
I see - the issue is less w/ the params into the `create`, but the fact that the create must return a `DendronEngineClient` and not an `EngineAPIService`. This causes the overridden version of `writeNote` to not get called. 

It still works ok now though because `engineClient` has a public constructor.